WooCommerce/Classes/ViewRelated/Upgrades/UpgradesView.swift

```diff
@@ -38,6 +38,10 @@ struct UpgradesView: View {
         ServiceLocator.stores.sessionManager.defaultSite?.name
     }
 
+    private var showingInAppPurchasesDebug: Bool {
+        ServiceLocator.generalAppSettings.betaFeatureEnabled(.inAppPurchases)
+    }
```

Comment:
I understand the logic for using the existing "Debug In-App Purchases" flag, as it reveals all the debug plans.

However, another way we could think of it is that this is an M2 flag, to start showing all the other plans as options. That way we can use it in future to switch the plan details UI out.

Doesn't need to change now, it's just a thought exercise – wdyt? Also, an M2 flag should probably enable only the 4 plans that we're going to have in M2, not all plans. E.g. we wouldn't want it to show multi-site plans if we add extra subscription groups.

Reply:
Logged it here: https://github.com/woocommerce/woocommerce-ios/issues/9948